### PR TITLE
Add a cime_env environment

### PR DESCRIPTION
**Pull request by jhkennedy**
In the CMDV-SM project, we've added some tests to the E3SM/CIME `SystemTests` that we'd like to put into production, but they require some of the scientific python dependencies and the `evv4esm` package developed for these test, which breaks the standard-lib only CIME paradigm. So, we'd like a lightweight python env. deployed to the E3SM supported machines that can be access and run on the compute nodes. This provides a lightweight `e3sm_simple` environment that includes the python packages CMDV-SM requires. 

This request has been supported more broadly, and CIME will be moving away from the standard lib only paradigm here soon as well, so a lightweight or simplified E3SM conda environment is going to become required in the future. We expect this first version will grow as people request more packages in a similar manner to `e3sm_unified`

----

This PR:
* moves the `e3sm_unified` build recipe into a `recipes/e3sm_unified` sub-directory, similar to how `conda-forge/staged-recipes` is structured (required because `conda build` requires a `meta.yaml` for each recipe and an alternate file name cannot be used).
* add a `e3sm_simple` metapackage build recipe in `recipes/e3sm_simple`
* adds testing of the `e3sm_simple` environment in travis-ci via singularity (@rpwagner, can you make sure I did this well?)

### CODE REVIEW

File: recipes/e3sm-simple/build.bash [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+#conda build -c conda-forge -c defaults -c e3sm -c cdat/label/v81 meta.yaml
```
Comment (xylar) [inline recipes/e3sm-simple/build.bash]:
```suggestion
conda build -c conda-forge -c defaults meta.yaml
```

Comment (xylar) [reply recipes/e3sm-simple/build.bash]: This probably shouldn't be commented out by default (I know you're following my lead), but I shouldn't have committed it commented out.

File: recipes/e3sm-simple/build.bash [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+upload=True
+version=1.0.0
+pythons=(27 36 37)
+build=0
+
+if [ $upload == "True" ]
+then
+   for python in "${pythons[@]}"
+   do
+      anaconda upload -u e3sm /home/${USER}/miniconda3/conda-bld/linux-64/e3sm-simple-${version}-py${python}_${build}.tar.bz2
```
Comment (xylar) [inline recipes/e3sm-simple/build.bash]:
```suggestion
      anaconda upload -u e3sm ${HOME}/miniconda3/conda-bld/linux-64/e3sm-simple-${version}-py${python}_${build}.tar.bz2
```

File: recipes/e3sm-unified/build.bash [hunk 3]
```diff
@@ -11,6 +11,6 @@ if [ $upload == "True" ]
 then
    for python in "${pythons[@]}"
    do
-      anaconda upload -u e3sm /home/xylar/miniconda3/conda-bld/linux-64/e3sm-unified-${version}-py${python}_${build}.tar.bz2
+      anaconda upload -u e3sm /home/${USER}/miniconda3/conda-bld/linux-64/e3sm-unified-${version}-py${python}_${build}.tar.bz2
```
Comment (xylar) [inline recipes/e3sm-unified/build.bash]:
```suggestion
      anaconda upload -u e3sm ${HOME}/miniconda3/conda-bld/linux-64/e3sm-unified-${version}-py${python}_${build}.tar.bz2
```

Comment (xylar) [reply recipes/e3sm-unified/build.bash]: Maybe uncomment the conda build command while you're at it.

File: recipes/e3sm-simple/build.bash [hunk 4]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+#conda build -c conda-forge -c defaults -c e3sm -c cdat/label/v81 meta.yaml
+
+upload=True
```
Comment (xylar) [inline recipes/e3sm-simple/build.bash]: Maybe safer if `upload=False` by default
```suggestion
upload=False
```

File: recipes/e3sm-unified/build.bash (hunk 3 above)
Comment (xylar) [reply recipes/e3sm-unified/build.bash]: Maybe set `upload=False` here as well

File: recipes/e3sm-simple/build.bash (hunk 4 above)
Comment (jhkennedy) [reply recipes/e3sm-simple/build.bash]: Done! See dcd9ac9c966474cc3132dccabbc8032fecd2ec87

File: recipes/e3sm-unified/build.bash (hunk 3 above)
Comment (jhkennedy) [reply recipes/e3sm-unified/build.bash]: Done! See dcd9ac9c966474cc3132dccabbc8032fecd2ec87

File: recipes/e3sm-simple/build.bash (hunk 2 above)
Comment (jhkennedy) [reply recipes/e3sm-simple/build.bash]: Done! See dcd9ac9c966474cc3132dccabbc8032fecd2ec87

File: recipes/e3sm-simple/build.bash (hunk 1 above)
Comment (jhkennedy) [reply recipes/e3sm-simple/build.bash]: Done! See dcd9ac9c966474cc3132dccabbc8032fecd2ec87

**Comment by jhkennedy** [comment]
@xylar Thanks for the suggestions! I did build and upload python 2.7, 3.6, and 3.7 versions of `e3sm_simple` to the anaconda.org/e3sm channel, so I think you should be able to do the cori install now.

**Comment by xylar** [comment]
Great, installing a python 2.7 and 3.7 env. on Cori right now.  I'll let you know when it's there.

**Comment by xylar** [comment]
@jhkennedy, both environments are there.  The chmod is still happening and will likely take awhile, so don't do anything silly like trying to install your own packages until that finishes ;-)  But other than that, you should be good to go.

**Comment by jhkennedy** [comment]
Awesome! Thanks for your help @xylar , especially on the weekend!

**Comment by xylar** [comment]
No problem. Greetings from PHL, on my way from Berlin to NM (via AZ).

**Comment by xylar** [comment]
@jhkennedy, I'm needing to make a new `e3sm-unified` soon.  Could we get this renamed and merged?  Or do you anticipate further changes and need more review?

**Comment by jhkennedy** [comment]
@xylar How soon? I've got a new EVV release in the pipe and would like to update this env. with that release -- I could probably have that together today and the new package at least queued on conda forge today. 

Outside of the rename and next EVV version I don't expect any other changes or need more review.

**Comment by xylar** [comment]
Not that soon. Waiting on processflow. Take your time.

**Comment by jhkennedy** [comment]
@xylar looks like the Travis-CI builds failed because of problems with singularity, as was true for de9ba189af96958484c0a227e2c638e645164a76. Should we try to get that fixed?

**Comment by jhkennedy** [comment]
@xylar I changed the name to `cime_systests` as suggested by @rljacob . Another name could be `cime_env` and then on the supported machines you'd end up with `source /path/to/load_latest_cime_env.sh` which might be the most transparent name, especially when this env. is going to be come the default python environment for the nightly testing on supported machines.

**Comment by xylar** [comment]
@jhkennedy, I think `cime_env` does sound better to me but I would defer to @rljacob (who I believe is away right now).

I wouldn't worry about the failing CI.  I won't be able to fix that myself and we'll need help from @rpwagner, but it's not worth holding up this PR if he doesn't have time to look at it right now.

**Comment by jhkennedy** [comment]
@xylar I'm going to change this to `cime_env`. Rob wanted the environment to be descriptive of what it is for, and while this is only used for *some* of the system tests, we can expect more parts of py-cime to use the scientific python ecosystem -- CIME developers at NERSC have indicated that they are moving to a standard load a python env. workflow for CIME as well as other people are adding in external python dependencies. So, `cime_env` is the most descriptive and robust over time name.

**Comment by jhkennedy** [comment]
@xylar Okay, I think we're g2g now! If you could deploy this alongside your next deployment of e3sm_unified, that'd be great! 

After that, I'll update the Jenkins jobs and then you can also delete the `e3sm_simple_1.0.0*` envs off of cori if you want (you'll get your `e3sm_unified` tab complete back!)

**Comment by xylar** [comment]
Great!  Let's just merge for now and if the scripts or something needs tweaking, I can do that as part of deploying e3sm-unified 1.3.0.

**Comment by jhkennedy** [comment]
Fantastic! Thanks @xylar

**Pull request merged**